### DISCUSSION TITLE
Help Needed 🙏 - Compatibility and Communication Issues with SolaX Inverter X1 Boost G4 (XB4)

### DISCUSSION THREAD

**Discussion by adilld** [wills106/homeassistant-solax-modbus#640] [Q&A]
Hello GitHub Community,

I would like to express my gratitude to [Wills106](https://github.com/wills106) for his outstanding work and continuous contributions to the HA community.

I hope this message finds you well. I am currently facing some challenges while working on the integration of a SolaX inverter, specifically the X1 Boost G4 series (XB4). I am reaching out to seek guidance and assistance regarding the following issues:

1. **Compatibility with SolaX Inverter X1 Boost G4 Series (XB4):**

I am uncertain about the compatibility of the integration with the SolaX inverter X1 Boost G4 series (XB4). If anyone has experience with this particular inverter model, I would appreciate any insights or advice on successful integration.

2. **Communication Issues:**

While attempting to communicate with the inverter, I encountered the following error: "SolaX: cannot find serial number, even not for MIC. unrecognized inverter type - serial number: unknown."

I suspect that the issue might be related to the Modbus USB RS485 interface. To troubleshoot this, I want to perform a basic communication test using a terminal emulation software like PuTTY on the host machine. However, I need guidance on the commands to send and what responses to expect from the SolaX inverter.

I would be grateful for any assistance, advice, or recommendations from the community regarding the mentioned issues. If you have experience with the SolaX G1 Boost series (XB4) or Modbus communication with SolaX inverters in general, your insights would be invaluable.

Thank you in advance for your time and support.

Best regards,
adilld

**Comment by ImadLTD** [comment]
+1

**Comment by adilld** [comment]
@wills106 can you check that discussion please ? Thanks in advance
